core/src/main/scala/org/apache/spark/scheduler/TaskResultGetter.scala

```diff
@@ -82,7 +84,19 @@ private[spark] class TaskResultGetter(sparkEnv: SparkEnv, scheduler: TaskSchedul
               (deserializedResult, size)
           }
 
-          result.metrics.setResultSize(size)
+          // Set the task result size in the accumulator updates received from the executors.
+          // We need to do this here on the driver because if we did this on the executors then
+          // we would have to serialize the result again after updating the size.
+          result.accumUpdates = result.accumUpdates.map { a =>
+            if (a.name == InternalAccumulator.RESULT_SIZE) {
+              assert(a.update.getOrElse(0L) == 0L,
```

Comment:
actually it should be equal to Some(0L). Maybe I should just explicitly write that.